jcl/src/jdk.jcmd/share/classes/openj9/tools/attach/diagnostics/info/JvmInfo.java

```diff
@@ -0,0 +1,168 @@
+/*[INCLUDE-IF Sidecar18-SE]*/
+/*******************************************************************************
+ * Copyright (c) 2019, 2019 IBM Corp. and others
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+ * or the Apache License, Version 2.0 which accompanies this distribution and
+ * is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following
+ * Secondary Licenses when the conditions for such availability set
+ * forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+ * General Public License, version 2 with the GNU Classpath
+ * Exception [1] and GNU General Public License, version 2 with the
+ * OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] http://openjdk.java.net/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0 WITH Classpath-exception-2.0 OR LicenseRef-GPL-2.0 WITH Assembly-exception
+ *******************************************************************************/
+package openj9.tools.attach.diagnostics.info;
+
+import java.io.IOException;
+import java.util.Set;
+
+import javax.management.openmbean.ArrayType;
+import javax.management.openmbean.CompositeData;
+import javax.management.openmbean.CompositeDataSupport;
+import javax.management.openmbean.CompositeType;
+import javax.management.openmbean.OpenDataException;
+import javax.management.openmbean.OpenType;
+import javax.management.openmbean.SimpleType;
+
+import com.ibm.java.lang.management.internal.LockInfoUtil;
+import com.ibm.java.lang.management.internal.MonitorInfoUtil;
+import com.ibm.java.lang.management.internal.StackTraceElementUtil;
+
+import openj9.tools.attach.diagnostics.base.DiagnosticProperties;
+
+/**
+ * Methods for converting between properties and CompositeTypes
+ *
+ */
+public class JvmInfo {
+	protected static final String ARRAY_SIZE = "array_size"; //$NON-NLS-1$
+	protected static void addFields(DiagnosticProperties props, String keyPrefix,
+			final CompositeType compType, CompositeData compData) {
+		for (String k: compType.keySet()) {
+			final String keyString = keyPrefix + k;
+			Object value = compData.get(k);
```

Comment:
Why is this local not `final` (like the others)?

Reply:
Good idea.  Fixed a number of other cases too.